models/targeting/genome_classes.py

```diff
@@ -212,3 +213,74 @@ def initialize_target_cut_probabilities(self, dt):
             for key_target in target_dict[key_domain].keys():
                 target = target_dict[key_domain][key_target]
                 target.compute_and_assign_cut_probability(dt)
+
+    def make_new_genome(self, indel_location, indel_size, new_genome):
+        """Re-index all domains and targets after single indel
+        and set current_genome to new_genome
+
+        This is a bit headache inducing
+        """
+        deleted_domains = []
+        broken_targets = []
+        target_dict = self.get_targets_from_genome()
+        for key_domain in target_dict.keys():
+            domain = self.domains[key_domain]
```

Comment:
Are you trying to use this `domain` as a copy to store some history? If so, we need to guarantee it's not acting as a pointer (I'm not sure), there are copy methods to correct it if it's a problem.